src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/AmendmentToEdifactMapper.java

```diff
@@ -0,0 +1,61 @@
+package uk.nhs.digital.nhsconnect.nhais.translator.amendment;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.FhirValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBody;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public abstract class AmendmentToEdifactMapper {
+
+    private static final String REMOVE_INDICATOR = "%";
+
+    protected static boolean amendmentPatchRequiringValue(AmendmentPatch amendmentPatch) {
+        return amendmentPatch.getOp() == AmendmentPatchOperation.ADD
+            || amendmentPatch.getOp() == AmendmentPatchOperation.REPLACE;
+    }
+
+    public List<Segment> map(AmendmentBody amendmentBody) throws FhirValidationException {
+        var patches = amendmentBody.getJsonPatches();
+        validatePatches(patches);
+        return mapAllPatches(patches);
+    }
+
+    protected void validatePatches(JsonPatches patches) throws FhirValidationException {
+    }
+
+    protected abstract List<Segment> mapAllPatches(JsonPatches patches);
```

Comment:
I'm not aware of a scenario where we'd need a mapper to return more than one segment